### PR TITLE
Gemma: Add logit soft-capping to score function.

### PR DESCRIPTION
**Pull request by RyanMullins**
Gemma 2 added logit soft-capping to `.call_with_cache()` in #1673 but this was not paralleled in the `.score()` function, so the logits/loss and derived attributes (e.g., gradients) will differ from those returned by `.generate()`. This PR brings these back into parity.

### CODE REVIEW

**Review by mattdangerw — APPROVED** [review]
lgtm! will pull in when tests finish
